### PR TITLE
Add Ruff

### PR DESCRIPTION
**Pull request by marco-carvalho**
## Overview
This PR introduces the integration of Ruff, a fast and efficient Python linter, into the `yfinance` project. The goal is to improve and maintain the code quality by enforcing style and coding standards.

## Benefits
- **Speed:** Ruff's performance-oriented design allows for quick linting, making it an ideal tool for continuous integration environments.
- **Readability and Maintenance:** Enforcing a consistent coding style enhances code readability and maintainability.
- **Error Prevention:** Early detection of potential errors and code smells helps in maintaining the overall quality of the codebase.

## Changes
- Applied Ruff checks to the existing codebase and made necessary adjustments to ensure compliance with the linter's rules.
- Created a new GitHub Action `ruff.yml` that triggers on pull requests to the `master` and `main` branches.

### CODE REVIEW

**Comment by ValueRaider** [comment]
This is much needed, thanks. But can you target `dev` branch instead? #1084 Just in case this broke something.

**Comment by ValueRaider** [comment]
@marco-carvalho I've rebased. Looks good to me, agree?

**Review by ValueRaider — COMMENTED** [review]
question

File: tests/prices.py [hunk 1]
```diff
@@ -270,60 +271,6 @@ def test_dailyWithEvents_bugs(self):
...
-
-            df_intraday_divs = df_intraday["Dividends"][df_intraday["Dividends"] != 0]
-            df_intraday_divs.index = df_intraday_divs.index.floor('D')
-            self.assertTrue(df_daily_divs.equals(df_intraday_divs))
-
-            test_run = True
-
-        if not test_run:
-            self.skipTest("Skipping test_intraDayWithEvents_tase() because no tickers had a dividend in last 60 days")
-
```
[the remark this answers is not in the document]
Comment (marco-carvalho) [reply tests/prices.py:326]: they are repeated

**Pull request merged**